Src/FluentAssertions/Equivalency/EquivalencyValidator.cs

```diff
@@ -64,12 +65,13 @@ private void RunStepsUntilEquivalencyIsProven(Comparands comparands, IEquivalenc
     {
         using var _ = context.Tracer.WriteBlock(node => node.Description);
 
+        Func<IEquivalencyStep, GetTraceMessage> getMessage = step => _ => $"Equivalency was proven by {step.GetType().Name}";
```

Comment:
In develop the lambda is constructed in _each_ iteration no matter if we enter the `if` or not.

See the difference on
[sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4oDchhGAzKQCYwA2MA5gIYAuM5OMgHEYPACpQuYGAFkYAZ3lcOMABQRgAKxhgepAHYQWAShoE6jCknJZSEqTCiEA3oVLvylnNYwpSAdSgASz4AGSD9NRFxSWk5RWV+AH1jZwBfQgzzIktbABEYADc2CAAHFzcPBnI/ADEICFVw+R4AHgieAD5SViCW40r3VwIPUdIAM2gYKQALUlVCrihSEJgAWxX9Hr6eAZGxocGDjyDx+dWNgF5L0hQsPeOx4cfH+2koADpA1fDI1X5Lt0Lh8xBAAMo8YL6DiqYymI6PLKIo5ZLIWGykAAKACUKvt3NVfKR6o1mm0Ot1ev0js9jhgAKztfQ8AA0pGib1kCiUKm64wArvowKQbkERd0kuKViDwZCIjD4fixpNYLN5otlhdNttqUrRrSXiszqotddbvcEccDYaPJzPt8whE1AKhSa+Gs4WYbaQkcdfe5UUA=)